src/main/java/com/nha/abdm/wrapper/hip/hrp/share/TokenNumberGenerator.java

```diff
@@ -0,0 +1,28 @@
+/* (C) 2024 */
+package com.nha.abdm.wrapper.hip.hrp.share;
+
+import java.time.LocalDate;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TokenNumberGenerator {
+
+  private AtomicInteger counter = new AtomicInteger(0);
+  private LocalDate currentDate = LocalDate.now();
+
+  @Scheduled(cron = "0 0 0 * * *")
+  public void resetTokenCount() {
```

Comment:
can you explain the logic behind generation of token? also, what does that cron expression evaluate to?

Reply:
the 0 0 0 * * * indicates.. seconds minutes hours days months years, so according to it the first 3 zeros indicate 12 am and the * indicate every day of month.